Automattic/wp-calypso
# Adding new useMyDomain flow to onboarding

#### Changes proposed in this Pull Request

This PR updates the onboading `use-your-domain` step to use the new flow already in use from the domain suggestions pages.

When selecting either the "Do you own it?" or the "Use a domain I own" links, the user will now be taken to the new steps to enter a domain and select whether they want to connect the domain or transfer it.

#### Testing instructions

Make sure that the flow works from both the `/start/domains` page and from the `/domains/add/use-my-domain` page (where it is already in use).

The pages should look something like this (depending on whether the selected domain is transferrable or not):
<img width="1792" alt="Screen Shot 2021-09-22 at 1 18 27 PM" src="https://user-images.githubusercontent.com/1379730/134394384-b9f8695c-73eb-45df-95d4-e1ac417583b0.png">


<img width="1792" alt="Screen Shot 2021-09-22 at 1 18 39 PM" src="https://user-images.githubusercontent.com/1379730/134394403-f3ae0227-6ed1-4ec8-8f9c-c9070aff3dd4.png">

Here is a screen recording showing selecting the domain for transfer:

https://user-images.githubusercontent.com/1379730/134394550-4e6249a8-cb9f-4d2e-98ab-9c87a9a07386.mov

And here is a screen recording showing selecting the domain for connecting (mapping):

https://user-images.githubusercontent.com/1379730/134394649-d0bd7137-6b41-4a36-88be-fabb5773a79e.mov
